bambi/backends/stan.py

```diff
@@ -74,11 +97,13 @@ def build(self, spec, reset=True):
         def _sanitize_name(name):
             ''' Stan only allows alphanumeric chars and underscore, and
             variable names must begin with a letter. So replace all invalid
-            chars with '_', prepend with 'b_', and track for later
+            chars with '_', prepend with 'b_' if needed, and track for later
             re-substitution. '''
             if name in self._original_names:
                 return name
-            clean = 'b_' + re.sub('[^a-zA-Z0-9\_]+', '_', name)
+            clean = re.sub('[^a-zA-Z0-9\_]+', '_', name)
+            if re.search(r'^\d', clean) is not None:
```

Comment:
This makes sense, but it turns out that Stan reserves a whole bunch of names that can't be used as variables, and I think it will probably be fairly common to run into these. E.g., the ESCS example won't run because 'floor' is a function name. So I'm going to revert to always prepending variable names, even if they don't start with numbers, just to be on the safe side.